backend/accounts.go

```diff
@@ -872,12 +874,42 @@ func (backend *Backend) createAndAddAccount(coin coinpkg.Coin, persistedConfig *
 		BtcCurrencyUnit:  backend.config.AppConfig().Backend.BtcUnit,
 	}
 
+	// This function is passed as a callback to the BTC account constructor. It is called when the
+	// keystore needs to determine whether an address belongs to an account on its same keystore.
+	getAddressCallback := func(askingAccount *btc.Account, scriptHashHex blockchain.ScriptHashHex) (*addresses.AccountAddress, bool, error) {
+		accountsByKeystore, err := backend.AccountsByKeystore()
+		if err != nil {
+			return nil, false, err
+		}
+		rootFingerprint, err := backend.keystore.RootFingerprint()
+		if err != nil {
+			return nil, false, err
+		}
+		for _, account := range accountsByKeystore[hex.EncodeToString(rootFingerprint)] {
+			// This only makes sense for BTC accounts.
+			btcAccount, ok := account.(*btc.Account)
+			if !ok {
+				continue
+			}
+			// Skip accounts that are not BTC or TBTC.
+			coinCode := btcAccount.Coin().Code()
+			if coinCode != coinpkg.CodeBTC && coinCode != coinpkg.CodeTBTC && coinCode != coinpkg.CodeLTC {
```

Comment:
I think the check you want is `if coinCode != askingAccount.Coin().Code() { continue }`. This is what I meant by

> You should probably also check that the addresses returned here are from the same account code as the account asking for the address.

After all, it doesn't make sense to e.g. return a Litecoin address here if the account asking is a Bitocoin account.

(also you needlessly excluded Litecoin Testnet :P)